LandSandBoat/server
# [core] [lua] [RUN] Implement Liement, add MDT/DT/Phalanx/Stoneskin checks where appropriate, general Absorb mechanic improvements

**_I affirm:_**

## What does this pull request do?

Implement liement, 

* add MDT/DT/Phalanx/Stoneskin checks where appropriate
* Adjust spikes damage to allow for absorbing
* Adjust spikes calls to use some actual message IDs instead of hardcodes
* Add core handler to check for Liement absorb
* Change some RUN effect removal to be silent
* Add range to Liement, which is not normally used unless you have Epeolatry's mod by design
* Update SteamJacket handler to pass in the element instead of hardcoded "ELEMENTAL" damage
* Adjust spell_damage.lua to have MDT + MDTII cap


## Steps to test these changes

Liement elemental absorb:
1. Fight fire elemental with blaze spikes, enfire
2. Use unda rune, use liement to absorb magic damage from nuke, enspell, spike damage. Each time Liement absorbs it should remove the buff from you. 

Liement AoE:
1. get a party
2. use Liement, it should be only activating on you
3. `!setmod LIEMENT_EXTENDS_TO_AREA 1` on yourself
4. use Liement, it should be AoE now

Dread spikes interaction:
1. find a mob with MP
2. force it to cast dread spikes `!exec target:castSpell(277, target)`
3. use Lux, liement
4. Hit mob
5. You should absorb your melee hit as HP, target will not absorb damage, see below:

![image](https://user-images.githubusercontent.com/60417494/172042645-6ad5d76c-ea20-406b-89c7-08bffb6987bf.png)

Magic weaponskill absorb:
1. equip staff
2. `!exec target:addStatusEffect(xi.effect.LIEMENT, 25, 0, 999, 0, xi.damageType.EARTH)` on target
3. observe absorb when using earth crusher

Hybrid weaponskill absorb:
1. equip great katana on SAM
2. use Tachi: Jinpu on target a couple times
3. `!exec target:addStatusEffect(xi.effect.LIEMENT, 25, 0, 999, 0, xi.damageType.WIND)` on target
4. use Tachi: Jinpu on target, repeat step 3 to get a few samples
5. Observe consistently lowered damage due to absorb

Blue Magic absorb:
1. change job to BLU
2. `!exec target:addStatusEffect(xi.effect.LIEMENT, 25, 0, 999, 0, xi.damageType.EARTH)` on target
3. observe absorb when using sandspin

SMN blood pact absorb (only works on nukes as most BPs are marked "physical" despite being hybrid):
1. change job to smn
2. summon ifrit
3. `!exec target:addStatusEffect(xi.effect.LIEMENT, 25, 0, 999, 0, xi.damageType.FIRE)` on target
7. observe absorb when using Fire BP rage

Helix absorb:
1. change job to SCH
2. `!exec target:addStatusEffect(xi.effect.LIEMENT, 25, 0, 999, 0, xi.damageType.ICE)` on target
3. observe absorb when using cryohelix
